### DISCUSSION TITLE
Cannot run/use OneDrive client on Ubuntu 22.04 after upgrade to 2.5.0

### DISCUSSION THREAD

**Discussion by cliverlong-1961** [abraunegg/onedrive#2820] [Q&A] [answered]
I was successfully using the OneDrive linux client to synchronize my personal OneDrive account files to my linux machine.

The linux Ubuntu OneDrive client prompted me to upgrade.

Clicking on the dialog link led to a missing page, [see](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md).

I followed the installation instructions [here](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204). I may have missed something during the installation - but I tried to follow the instructions to the letter.
```
Executing command: ~/.config/onedrive$ onedrive --version
```
I see message: onedrive v2.5.0-1+np2+2.1

Creating files or folders on the OneDrive webpage after the upgrade, these do not appear on the linux machine running the client.

When I run the following command: `onedrive --display-config`

I get the following messages:
```
D-Bus message bus daemon is available; GUI notifications are now enabled
Reading configuration file: /home/clive/.config/onedrive/config

Invalid value for key in config file: drive_id
```
I notice under ~/.config/onedrive I see files:

config and config_backup

and directory accounts that contains 
config and config_backup 

When I rename the config file using: mv config config_xx

Then run: `onedrive --dry-run`

The following messages are displayed:
```
D-Bus message bus daemon is available; GUI notifications are now enabled
DRY-RUN Configured. Output below shows what 'would' have occurred.
DRY-RUN: Copying items.sqlite3 to items-dryrun.sqlite3 to use for dry run operations
DRY RUN: Not creating backup config file as --dry-run has been used
DRY RUN: Not updating hash files as --dry-run has been used
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Configuring Global Azure AD Endpoints

Your command line input is missing either the '--sync' or '--monitor' switches. Please include one (but not both) of these switches in your command line, or refer to 'onedrive --help' for additional guidance.

It is important to note that you must include one of these two arguments in your command line for the application to perform a synchronisation with Microsoft OneDrive
```
Then run: `onedrive --dry-run -- sync`

The following messages are displayed:
```
D-Bus message bus daemon is available; GUI notifications are now enabled
DRY-RUN Configured. Output below shows what 'would' have occurred.
DRY-RUN: Copying items.sqlite3 to items-dryrun.sqlite3 to use for dry run operations
DRY RUN: Not creating backup config file as --dry-run has been used
DRY RUN: Not updating hash files as --dry-run has been used
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Configuring Global Azure AD Endpoints
Fetching items from the OneDrive API for Drive ID: eaa39626be95a5df ..
A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.

A database statement execution error occurred: NOT NULL constraint failed: item.type

<repeat>
```

If I rename config to config_yy, then run command: `onedrive --dry-run --resync`

I get following messages:
```
D-Bus message bus daemon is available; GUI notifications are now enabled
DRY-RUN Configured. Output below shows what 'would' have occurred.
DRY-RUN: No database copy created for --dry-run due to --resync also being used

The usage of --resync will delete your local 'onedrive' client state, thus no record of your current 'sync status' will exist.
This has the potential to overwrite local versions of files with perhaps older versions of documents downloaded from OneDrive, resulting in local data loss.
If in doubt, backup your local data before using --resync

Are you sure you wish to proceed with --resync? [Y/N] Y

DRY RUN: Not removing hash files as --dry-run has been used
DRY RUN: Not creating backup config file as --dry-run has been used
DRY RUN: Not updating hash files as --dry-run has been used
Deleting the saved application sync status ...
DRY RUN: Not removing the saved application sync status
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Configuring Global Azure AD Endpoints

Your command line input is missing either the '--sync' or '--monitor' switches. Please include one (but not both) of these switches in your command line, or refer to 'onedrive --help' for additional guidance.

It is important to note that you must include one of these two arguments in your command line for the application to perform a synchronisation with Microsoft OneDrive
```
Is there something I can fix in the config file to reconnect the OneDrive client to my personal OneDrive account?

**Comment by abraunegg** [comment, marked as answer]
@cliverlong-1961 
So lets start with the basics - please *read* the documentation: 
* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md
* https://github.com/abraunegg/onedrive/blob/master/docs/client-architecture.md
* https://github.com/abraunegg/onedrive/blob/master/docs/application-config-options.md

> Clicking on the dialog link led to a missing page, [see](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md).

What dialog link where?

I know that there is an issue here (https://abraunegg.github.io/) as this requires GitHub actions to be setup to re-deploy the site, but were is this 'dialog link' ???

> `Invalid value for key in config file: drive_id`

This is generally an issue if you have been using the OneDrive GUI - https://github.com/bpozdena/OneDriveGUI - have you installed / used this at all and have you upgraded your GUI to be compatible with onedrive v2.5.0?

The v2.5.0 checks for invalid 'config' file entries, and the GUI, at one point in time created all entries with default values - and this one blank - which cannot be blank ... which leads to this error message.

The way you resolve this is:
1. Delete this line
2. Edit | Comment this line out

> `onedrive --dry-run`

As per the documentation and the error message ... you *must* tell the client what you want to do .. you are missing those parameters. The error output is really clear on *why* you are getting this error message.

> Is there something I can fix in the config file to reconnect the OneDrive client to my personal OneDrive account?

Please *read* the documentation first.

Once you have done this, please run:

```
onedrive --sync --verbose --resync --resync-auth
```

This will clean up your database and your `NOT NULL constraint failed: item.type` and get you to a point of good health.

**Comment by cliverlong-1961** [comment]
As a link to this issue - I think the majority of issues were due to an old version of the OneDriveGUI client trying to communicate with OneDrive client 2.5. 1. I deleted as much as the "old" material as possible, 2. performed a OneDrive client --sync --resync of 66GB of date which ran overnight (I had issues with my network link now resolved) and 3. launched correct OneDriveGUI client making corrections to the config file settings - all described [here](https://github.com/bpozdena/OneDriveGUI/discussions/187).

Overall I think I had older versions of the two clients installed from Ubuntu repositories and the updated code mainly required a change to the Ubuntu startup of the OneDriveGUI client startup and the correct config file. I have provided the link and summary here if someone has a similar issues in future.
